File: base/animal2/.cache/cheat.java

class Animal {
    private String especie;
    private String barulho;
    private int estagio;

    public Animal(String especie, String barulho) {
        this.especie = especie;
        this.barulho = barulho;
        this.estagio = 0;
    }

    public String fazerBarulho() {
        if (estagio == 0) {
            return "---";
        }
        if (estagio == 4) {
            return "RIP";
        }
        return this.barulho;
    }

    public void envelhecer(int nivel) {
        estagio += nivel;
        if (estagio >= 4) {
            System.out.printf("warning: %s morreu%n", especie);
            estagio = 4;
        }
    }

    public String toString() {
        return String.format("%s:%d:%s", especie, estagio, barulho);
    }
}

public class Adapter {
    private Animal animal;
    public Adapter() {
        this.animal = new Animal("", "");
    }

    public void init(String especie, String barulho) {
        animal = new Animal(especie, barulho);
    }
    
    public String noise() {
        return animal.fazerBarulho();
    }

    public void grow(int nivel) {
        animal.envelhecer(nivel);
    }

    public String toString() {
        return animal.toString();
    }
}

File: base/animal2/.cache/cheat.ts

class Animal {
    especie: string;
    barulho: string;
    estagio: number;

    constructor(especie: string, barulho: string) {
        this.especie = especie;
        this.barulho = barulho;
        this.estagio = 0;
    }

    fazerBarulho(): string {
        if (this.estagio === 0) {
            return "---";
        }
        if (this.estagio === 4) {
            return "RIP";
        }
        return this.barulho;
    }

    envelhecer(nivel: number): void {
        this.estagio += nivel;
        if (this.estagio >= 4) {
            console.log(`warning: ${this.especie} morreu`);
            this.estagio = 4;
        }
    }

    toString(): string {
        return `${this.especie}:${this.estagio}:${this.barulho}`;
    }
}

class Adapter {
    animal: Animal;
    constructor() {
        this.animal = new Animal("", "");
    }

    init(especie: string, barulho: string): void {
        this.animal = new Animal(especie, barulho);
    }

    grow(nivel: number): void {
        this.animal.envelhecer(nivel);
    }

    noise(): string {
        return this.animal.fazerBarulho();
    }

    show(): string {
        return this.animal.toString();
    }
}

export {Adapter};
